_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,662 @@
+
+Our app release cadence is still about a month long, moving to React Native hasn't changed that. We've automated the entire process, so it's a cultural artifact rather than technical. So a day or two for the App Store review is fine, ideally our betas should be getting longer than that for testing.
+
+This is a little bit tricky though. Because you can write expose native code which interacts with your JavaScript, this creates a link between the native API that is available and the JavaScript that uses it. So, you still need to ship native code. 
+
+
+## Doing it right per platform 
+
+[TODO]
+
+-   RN gives you the ability to think in cross-platform
+-   Most devs are JS people trying to ship to Android/iOS
+-   They're interested in getting it done vs getting it right
+-   show my navigation issue
+
+-   Doing it right will inevitably require native code
+-   RN is a focused UI framework 
+-   Does not hinder you from making apps with traditionally native features: e.g. NSUserActivity, Spotlight etc
+-   Making some of these features requires crossing the bridge back to native land and having the work done there
+-   _We_ still think in terms of `UIViewControllers` which have a react view tree, not there is an app react component with sub-react components as view controllers.
+
+-   RN provides some pretty clever ways to handle cross-platform 
+
+-   note somewhere that our imageviews use SDWebImage to share cache
+
+## Create React Native App
+
+One of the biggest projects to happen in the React Native world in the last 6 months is Create React Native App (CRNA). This is the "super easy to get started" React Native experience. 
+
+Remember that most of the people coming to React Native are web developers, and the idea of writing Objective-C/Swift/Java to them is unappealing. CRNA actually removes the ability for you to write native code, and trades that with letting a company called Expo do it for you.
+
+Expo are a pretty new company, whose work is entirely open source. They've done a lot of interesting work in the community, and as individuals they are well respected for contributions to React Native the library - and a bunch of related libraries. You can use the Expo app from the App Store to run your CRNA project on any iOS/Android device instantly, and the app has a lot more baseline UI components to work with than React Native does on its own.
+
+With CRNA you are giving up control of the native side, but gaining a lot on the ease-of-use side. CRNA doesn't force the project to stay this way though, you can eject your app from the Expo and start adding native code to the app. 
+
+Why did I even mention this? Well, if you're looking at React Native for a greenfield app (e.g. something new), CRNA may be your best option. When you're getting started, less options is better, and this is the optimal setup according to the React Native team. 
+
+## Animations
```

Comment:
@alloy - I would particularly like your feedback from here downwards. I'm covering trade-offs that we've made for this, and want to make sure this matches your feelings too